### DISCUSSION TITLE
Switch animation keys using property

### DISCUSSION THREAD

**Discussion by AlexKiryanov** [friction2d/friction#391] [Q&A] [answered]
Hello @rodlie ! Here’s the question. Suppose I have a rectangle with keys in position property.
Frame 1 one x,y, frame 2 other x,y frame 3 other x,y
There is also a control object with property(numeric field.) in the scene.
Can I switch the numbers in the property of the control object, switch keys in the rectangle animation.
Number 1 animation keys from the first frame
Number 2 animation keys from second frame
Number 3 animation keys from the third frame
and so on
I hope I made my point correctly.

**Comment by AlmostMikhail** [comment]
Nice to see the legend(Alexander Kiryanov) of enve and friction on here.

**Comment by pgilfernandez** [comment, marked as answer]
> Hello @rodlie ! Here’s the question. Suppose I have a rectangle with keys in position property. Frame 1 one x,y, frame 2 other x,y frame 3 other x,y There is also a control object with property(numeric field.) in the scene. Can I switch the numbers in the property of the control object, switch keys in the rectangle animation. Number 1 animation keys from the first frame Number 2 animation keys from second frame Number 3 animation keys from the third frame and so on I hope I made my point correctly.

Umm, let me try to understand your question: you want to have some kind of switch (you don't specify how are you going to set it or animate it but I guess at this point it's not important) to control the position of an object based on positions saved on another object along different keyframes, right?
If that's correct then I bet the answer is not, you cannot do it as there is no any function (yet) that lets you grab a value from another time/frame and use it on the actual frame you want. It's a request some users have done (including me) to add a valueOnTime() or something like that.

But, if you don't mind having those positions written into an expression that switches them controlled by, for instance, a "custom property", then it's possible to do it.  [Here you have an example (switch_v01.friction.zip)](https://github.com/user-attachments/files/18234091/switch_v01.friction.zip).


Cheers

**Reply by AlexKiryanov** [reply]
Hi! Thank you very much for your reply!
I have looked at your file, unfortunately it is not what I wanted. In your file you write the position value, and then switch these position values according to the property value.
I need to switch exactly keys, that is, I created position keys for the object, then with the help of the switch property. these keys.
Something like time-remapping function with controller

**Reply by pgilfernandez** [reply]
Then I'm lost, I don't understand what do you want to do... could you maybe tell us what other software offers that feature and could you show us it in action?

Thanks

**Reply by AlexKiryanov** [reply]
[test.zip](https://github.com/user-attachments/files/18241945/test.zip)
Hi!
Something like this.
Here I control the time remapping function through a property in object CC.
Your idea basically works too, but one minus I have to enter the exact coordinates where I want to move my object. It’s the way out, but I had an idea how to switch between keys without using the time remapping function
![image](https://github.com/user-attachments/assets/20d674ef-b8cb-47bb-b837-720a10433a79)
Like in the terrible After Effects.

Can I ask a question. How do I get properties from other scenes?
Well, suppose I have to control one object with another object that is in another scene.
Example:
In Scene 1 I give the object an expression for position.  
my_var x = Scene02.CC_object.transform.translation.x;
But this option for some reason does not work....

**Reply by AlexKiryanov** [reply]
@rodlie Hi! Maybe you know how to take variables from another scene?

"Well, suppose I have to control one object with another object that is in another scene.
Example:
In Scene 1 I give the object an expression for position.
my_var x = Scene02.CC_object.transform.translation.x;
But this option for some reason does not work...."
